### PR TITLE
Add documentation for Cloudflare R2 usage.

### PR DESCRIPTION
**Pull request by MrAlex94**
R2 is S3-compatible. Here are some light docs on how to use it.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -28,6 +28,7 @@ Table of Contents (ToC)
 * Storage Options
   * [Local](docs/Local.md)
   * [S3](docs/S3.md)
+  * [R2](docs/S3.md#R2)
```
Comment (Xuanwo) [inline README.md:31]: I'm not sure if we need separate section for all s3-compatible services. Maybe inside `S3.md` is enough?

Leave this for maintainers @sylvestre @drahnr to decide.

File: docs/Configuration.md [hunk 2]
```diff
@@ -86,6 +86,13 @@ configuration variables
 The endpoint used then becomes `${SCCACHE_BUCKET}.s3-{SCCACHE_REGION}.amazonaws.com`.
 If `SCCACHE_REGION` is undefined, it will default to `us-east-1`.
 
+#### cloudflare r2
+
+* `SCCACHE_BUCKET` is the name of your R2 bucket.
+* `SCCACHE_ENDPOINT` must follow the format of `https://<ACCOUNT_ID>.r2.cloudflarestorage.com`. Note that the `https://` must be included. Your account ID can be found [here](https://developers.cloudflare.com/fundamentals/get-started/basic-tasks/find-account-and-zone-ids/).
+* (Optional) `SCCACHE_REGION` can be set to `auto` or `us-east-1`.
```
Comment (Xuanwo) [inline docs/Configuration.md:93]: This is required for `r2`.

File: docs/Configuration.md [hunk 3]
```diff
@@ -86,6 +86,13 @@ configuration variables
...
+* `SCCACHE_S3_USE_SSL` must be set to `true`.
```
Comment (Xuanwo) [inline docs/Configuration.md:94]: If we are using `https://` in endpoint, `SCCACHE_S3_USE_SSL` cloud be ignored.

File: README.md (hunk 1 above)
Comment (MrAlex94) [reply README.md:31]: Probably not needed here, but maybe worth changing `S3` to `S3 and compatible` or something similar?

Comment (Xuanwo) [reply README.md:31]: Nice idea to me.

Comment (sylvestre) [reply README.md:31]: sounds good

**Comment by MrAlex94** [comment]
Updated the S3 doc to be more informative and removed the unused env vars as well.

File: README.md (hunk 1 above)
Comment (Xuanwo) [reply README.md:31]: > maybe worth changing `S3` to `S3 and compatible` or something similar?

Please do this change, others LGTM~

**Pull request merged**